### PR TITLE
solved issue 2813

### LINKED ISSUE #2813: No Python bindings for cv::stereo::QuasiDenseStereo
##### System information (version)
- OpenCV => 4.4.0
- Operating System / Platform => Linux
- Compiler => Python PIP install (opencv-python==4.4.0.42)

##### Detailed description

There are no Python bindings for the QuasiDenseStereo module: https://docs.opencv.org/4.4.0/dc/df4/classcv_1_1stereo_1_1QuasiDenseStereo.html#details

##### Issue submission checklist
Also asked [OpenCV forum](https://forum.opencv.org/t/cant-find-python-bindings-for-cv-quasidensestereo/819).

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       answers.opencv.org, Stack Overflow, etc and have not found solution
   
 - [ ] I updated to latest OpenCV version and the issue is still there

### PR DESCRIPTION
**Pull request by LaurentBerger**
resolves #2813

<cut/>

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by LaurentBerger** [comment]
@alalek there is some problem with buildbot?

File: modules/stereo/include/opencv2/stereo/quasi_dense_stereo.hpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ class CV_EXPORTS QuasiDenseStereo
      * in case of video processing.
      * @sa loadParameters
      */
-    virtual int loadParameters(cv::String filepath) = 0;
+    CV_WRAP virtual int loadParameters(cv::String filepath) = 0;
```
Comment (alalek) [inline modules/stereo/include/opencv2/stereo/quasi_dense_stereo.hpp:116]: Looks like bindings scripts can't properly wrap abstract classes and pure methods.

File: modules/stereo/samples/sample_quasi_dense.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import numpy as np
+import cv2 as cv
+
+left_img = cv.imread("G:/Lib/opencv/samples/data/aloeL.jpg", cv.IMREAD_COLOR)
+right_img = cv.imread("G:/Lib/opencv/samples/data/aloeR.jpg", cv.IMREAD_COLOR)
```
Comment (alalek) [inline modules/stereo/samples/sample_quasi_dense.py:5]: Please remove absolute paths. Use command-line parameters with default values.
Use `cv.samples.findFile()`

File: modules/stereo/samples/sample_quasi_dense.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+stereo = cv.stereo.QuasiDenseStereo_create(frame_size[::-1])
+stereo.process(left_img,right_img)
+disp = stereo.getDisparity(80)
+cv.imshow("disparity", disp)
+cv.waitKey()
+dense_matches = stereo.getDenseMatches()
+try:
+    f = open("dense.txt", "wt")
+    with f:
```
Comment (alalek) [inline modules/stereo/samples/sample_quasi_dense.py:17]:
```
with open("dense.txt", "wt") as f:
```

**Review by alalek — COMMENTED** [review]
Well done!

**Review by alalek — APPROVED** [review]
@LaurentBerger Well done! Thank you for contribution :+1:

**Comment by ZhuLingfeng1993** [comment]
@LaurentBerger 
Have you test the correctness of quasi dense stereo algorithm?
https://github.com/opencv/opencv_contrib/pull/1941#issuecomment-778613997

**Pull request merged**